Makefile

```diff
@@ -12,27 +14,40 @@
 #See the License for the specific language governing permissions and
 #limitations under the License.
 
-CXXFLAGS = -c -std=c++17 -O2 -Wall -Wextra -Werror -g
+ifeq (${DEBUG},1)
+	CXXFLAGS = -std=c++17 -O0 -Wall -Wextra -Werror -g
+else
+	CXXFLAGS = -std=c++17 -O2 -Wall -Wextra -Werror
+endif
```

Comment:
May I recommend:

```make
CXXFLAGS = -std=c++17 -Wall -Wextra -Werror
ifdef DEBUG
CXXFLAGS += -g -O0
else
CXXFLAGS += -O2
endif
```

Avoids duplicating the non-conditional flags. Also indenting conditional bodies in Make is not conventional, because indentation has very specific rules in Make.

Reply:
I don't like `ifdef DEBUG` because it treats the condition as true in the case of `make DEBUG=0` . Do you know of any way of writing a conditional that will hold whenever DEBUG is "truthy" ?

Otherwise I think the selection is fine.

Reply:
Nope. Makefiles suck.

Reply:
Got it. Then I will stick with `ifeq (${DEBUG},1)` , and apply the other points of your suggestion, if that is OK with you.